code/parachain/frame/cosmwasm/src/hook.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{runtimes::vm::CosmwasmVM, types::*, Config, Error};
+use cosmwasm_vm::{
+	cosmwasm_std::{ContractResult, QueryResponse, Response},
+	vm::{VMBase, VmErrorOf},
+};
+use cosmwasm_vm_wasmi::WasmiVM;
+
+/// A hook into the VM, used to call precompiled contracts.
+pub trait Hook<T: Config> {
```

Comment:
Nit: a hook doesn't necessarily need to forward to precompiles; we can split this into two traits:

PrecompiledInfo + Hook

and remove the precompiled_prefix from member functions.